system/HTTP/Cookie/CookieInterface.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HTTP\Cookie;
+
+/**
+ * Interface for a value object representation of an HTTP cookie.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+ */
+interface CookieInterface
+{
+	/**
+	 * Cookies will be sent in all contexts, i.e in responses to both
+	 * first-party and cross-origin requests. If `SameSite=None` is set,
+	 * the cookie `Secure` attribute must also be set (or the cookie will be blocked).
+	 */
+	public const SAMESITE_NONE = 'none';
+
+	/**
+	 * Cookies are not sent on normal cross-site subrequests (for example to
+	 * load images or frames into a third party site), but are sent when a
+	 * user is navigating to the origin site (i.e. when following a link).
+	 *
+	 * @var string
+	 */
+	public const SAMESITE_LAX = 'lax';
+
+	/**
+	 * Cookies will only be sent in a first-party context and not be sent
+	 * along with requests initiated by third party websites.
+	 *
+	 * @var string
+	 */
+	public const SAMESITE_STRICT = 'strict';
+
+	/**
+	 * RFC 6265 allowed values for the "SameSite" attribute.
+	 *
+	 * @var string[]
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
+	 */
+	public const ALLOWED_SAMESITE_VALUES = [
+		self::SAMESITE_NONE,
+		self::SAMESITE_LAX,
+		self::SAMESITE_STRICT,
+	];
+
+	/**
+	 * Expires date format.
+	 *
+	 * @var string
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date
+	 * @see https://tools.ietf.org/html/rfc7231#section-7.1.1.2
+	 */
+	public const EXPIRES_FORMAT = 'D, d-M-Y H:i:s T';
```

Comment:
expires format and getExpiresString are not used by setcookie, of course. This is used if you want to pass the set-cookie header directly using `header()`

Reply:
@paulbalandan it's unused constant, use `DateTimeInterface::COOKIE`

Reply:
This is used in `Cookie::__toString()`

Reply:
`DateTimeInterface::COOKIE` is `l, d-M-Y H:i:s T`. But the `Set-Cookie` headers accepts the Expires date using the short form of the day, i.e `Mon` instead of `Monday`.

Reply:
@paulbalandan nope,
~~~
Sun, 06 Nov 1994 08:49:37 GMT  ; RFC 822, updated by RFC 1123
Sunday, 06-Nov-94 08:49:37 GMT ; RFC 850, obsoleted by RFC 1036
Sun Nov  6 08:49:37 1994       ; ANSI C's asctime() format
~~~

Reply:
Those three are supported but the last two are obsolete already. If you will read the next paragraph of RFC 7231 after enumerating those formats, it says recipients must accept all formats.

>  A recipient that parses a timestamp value in an HTTP header field
   MUST accept all three HTTP-date formats.  When a sender generates a
   header field that contains one or more timestamps defined as
   HTTP-date, the sender MUST generate those timestamps in the
   IMF-fixdate format.

BUT looking at the last sentence it says you need to generate the timestamp into IMF-fixdate format, which is actually the first format you listed as supported. So why go the extra mile of using obsolete formats then go change it to the current format if I can start with the current format already?

Reply:
Further reading: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date